Scripts/Core/Contexts/EditingContextLauncher.cs

```diff
@@ -0,0 +1,227 @@
+﻿#if UNITY_EDITOR && UNITY_EDITORVR
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.InputNew;
+
+namespace UnityEditor.Experimental.EditorVR.Core
+{
+	class EditingContextLauncher : MonoBehaviour
+	{
+		const string k_DefaultContext = "EditorVR.DefaultContext";
+
+		internal static EditingContextLauncher s_Instance;
+		static InputManager s_InputManager;
+
+		List<IEditingContext> m_ContextStack = new List<IEditingContext>();
+
+		List<IEditingContext> m_AvailableContexts = new List<IEditingContext>();
+		string[] m_ContextNames;
+		int m_SelectedContextIndex;
+
+		IEditingContext defaultContext
+		{
+			get
+			{
+				var context = m_AvailableContexts.First();
+
```

Comment:
The defaultContextName block is only for whether there is a default context saved. I don't think that the context var initialization is part of that. Happy to discuss if you don't agree.